jobs/Llama2/finetune/eval_7b_hf.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from peft import AutoPeftModelForCausalLM
+from transformers import AutoTokenizer
+
+model_dir = "/home/cindyli/projects/ctb-whkchun/s2_bliss/results-finetune-7b-hf-3epochs/checkpoint-975"
+
+# load base LLM model and tokenizer
+model = AutoPeftModelForCausalLM.from_pretrained(
+    model_dir,
+    low_cpu_mem_usage=True,
+    torch_dtype=torch.float16,
+    load_in_4bit=True,
+    local_files_only=True
+)
+tokenizer = AutoTokenizer.from_pretrained(model_dir, local_files_only=True)
+
+
+# Evaluation
+# Inference. Generated texts by giving an instruction
+def generate_text_with_instruction(instruction, input, model, tokenizer, temperature=0.7):
+    input_key = "### Input:\n"
+    response_key = "### Response:\n"
+    prompt = f"{instruction}{input_key}{input}\n\n{response_key}\n"
+    input_ids = tokenizer(prompt, return_tensors="pt", truncation=True).input_ids.cuda()
+    print(f"Instruction: {instruction}\n")
+    print(f"Prompt: {input}\n")
+    outputs = model.generate(input_ids=input_ids, max_new_tokens=100, do_sample=True, top_p=0.9, temperature=temperature)
+    print(f"Generated instruction (temprature {temperature}): {tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0][len(prompt):]}\n\n")
+
+
+def generate_text_with_prompt(prompt, model, tokenizer, temperature=0.7):
+    input_ids = tokenizer(prompt, return_tensors="pt", truncation=True).input_ids.cuda()
+    print(f"Prompt: {prompt}\n")
+    outputs = model.generate(input_ids=input_ids, max_new_tokens=100, do_sample=True, top_p=0.9, temperature=temperature)
+    print(f"Generated instruction (temprature {temperature}): {tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0][len(prompt):]}\n\n")
+
+
+# Test with exactly same instructions used in fine-tuning
```

Comment:
If the instruction is _exactly_ the same as used in fine-tuning, it would be safer to have a common definition that is shared between the fine-tuning script and this evaluation script (and any other scripts).  However, in the [fine-tuning script](https://github.com/inclusive-design/baby-bliss-bot/blob/4cf061c6f0505a790d452ccc89c531bbba45205e/jobs/Llama2/finetune/finetune_7b_hf.py#L123) the instruction is constructed differently depending on a `direction` parameter (`EnglishToBliss` vs. `BlissToEnglish`).  A way to do it is to have a `settings.py` file that includes an `INSTRUCTIONS` dictionary where `EnglishToBliss` and `BlissToEnglish` are keys to the two prompts.  But, maybe that's overkill -- what do you think?